src/ComponentBase.ts

```diff
@@ -82,6 +82,7 @@ abstract class ComponentBase<P extends React.Props<any>, S extends Object> exten
     private _handledSubscriptionsLookup: SubscriptionLookup<S> = {};
 
     private _isMounted = false;
+    private _nextState: S;
```

Comment:
Should you update `_nextState` in more places? Right now, it is only the initial state, not really the 'next' state.